tools/Microsoft.TemplateEngine.Authoring.CLI/Commands/validate/ValidateCommand.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Settings;
+
+namespace Microsoft.TemplateEngine.Authoring.CLI.Commands
+{
+    internal class ValidateCommand : ExecutableCommand<ValidateCommandArgs>
+    {
+        private const string CommandName = "validate";
+
+        private readonly Argument<string> _templateLocationArg = new("template-location")
+        {
+            Description = "The location of template(s) to validate",
+            // 0 for case where only path is specified
+            Arity = new ArgumentArity(1, 1)
+        };
+
+        public ValidateCommand() : base(CommandName, "Validates the templates at given location.")
+        {
+            AddArgument(_templateLocationArg);
+        }
+
+        protected internal override ValidateCommandArgs ParseContext(ParseResult parseResult)
+        {
+            return new ValidateCommandArgs(parseResult.GetValue(_templateLocationArg));
+        }
+
+        protected override async Task<int> ExecuteAsync(ValidateCommandArgs args, ILoggerFactory loggerFactory, CancellationToken cancellationToken)
+        {
+            ILogger logger = loggerFactory.CreateLogger(CommandName);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            using IEngineEnvironmentSettings settings = SetupSettings(loggerFactory);
+            Scanner scanner = new(settings);
+
+            logger.LogInformation("Scanning location '{0}' for the templates...", args.TemplateLocation);
+
+            ScanResult scanResult = await scanner.ScanAsync(
+                args.TemplateLocation,
+                logValidationResults: false,
+                returnInvalidTemplates: true,
+                cancellationToken).ConfigureAwait(false);
+            cancellationToken.ThrowIfCancellationRequested();
+            logger.LogInformation("Scanning completed");
+            PrintResults(logger, scanResult);
+            return 0;
+        }
+
+        private IEngineEnvironmentSettings SetupSettings(ILoggerFactory loggerFactory)
+        {
+            var builtIns = new List<(Type InterfaceType, IIdentifiedComponent Instance)>();
+            builtIns.AddRange(Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Components.AllComponents);
+            builtIns.AddRange(Microsoft.TemplateEngine.Edge.Components.AllComponents);
+
+            ITemplateEngineHost host = new DefaultTemplateEngineHost("template-validator", "1.0", builtIns: builtIns, loggerFactory: loggerFactory);
```

Comment:
could you say what "1.0" is about? can it be changed?

Reply:
We need to give the version when creating the host, in this case it can be anything as settings are virtualized.